webknossos/webknossos/dataset/view.py

```diff
@@ -29,53 +29,127 @@ def __init__(
-        # Adding the bounding box as parameter is a workaround for this.
-        # However, keep in mind that this bounding box is just a snapshot.
-        # This bounding box is not updated if the bounding box of the dataset is updated.
-        # Even though an outdated bounding box can lead to missed (or unwanted) warnings,
-        # this is sufficient for our use case because such scenarios are unlikely and not critical.
-        # This should not be misused to infer the size of the dataset because this might lead to problems.
-        self._mag_view_bbox_at_creation = mag_view_bbox_at_creation
+        self._mag = mag
 
     @property
     def header(self) -> wkw.Header:
         return self._header
 
     @property
-    def size(self) -> Vec3Int:
-        return self._size
+    def bounding_box(self) -> BoundingBox:
+        assert self._bounding_box is not None
+        return self._bounding_box
+
+    @property
+    def mag(self) -> Mag:
+        return self._mag
 
     @property
     def global_offset(self) -> Vec3Int:
-        return self._global_offset
+        warnings.warn(
+            "[DEPRECATION] view.global_offset is deprecated. "
+            + "Since this is a MagView, please use "
+            + "view.bounding_box.in_mag(view.mag).topleft instead."
+        )
+        return self.bounding_box.in_mag(self._mag).topleft
+
+    @property
+    def size(self) -> Vec3Int:
+        warnings.warn(
+            "[DEPRECATION] view.size is deprecated. "
+            + "Since this is a MagView, please use "
+            + "view.bounding_box.in_mag(view.mag).size instead."
+        )
+        return self.bounding_box.in_mag(self._mag).size
 
     @property
     def read_only(self) -> bool:
         return self._read_only
 
+    def _get_mag1_bbox(
+        self,
+        abs_mag1_bbox: Optional[BoundingBox] = None,
+        rel_mag1_bbox: Optional[BoundingBox] = None,
+        abs_mag1_offset: Optional[Vec3IntLike] = None,
+        rel_mag1_offset: Optional[Vec3IntLike] = None,
+        mag1_size: Optional[Vec3IntLike] = None,
+        # current_mag_bbox: Optional[BoundingBox] = None,
+        abs_current_mag_offset: Optional[Vec3IntLike] = None,
+        rel_current_mag_offset: Optional[Vec3IntLike] = None,
+        current_mag_size: Optional[Vec3IntLike] = None,
+    ) -> BoundingBox:
+        num_bboxes = sum(i is not None for i in [abs_mag1_bbox, rel_mag1_bbox])
+        num_offsets = sum(
+            i is not None
+            for i in [
+                abs_mag1_offset,
+                rel_mag1_offset,
+                abs_current_mag_offset,
+                rel_current_mag_offset,
+            ]
+        )
+        num_sizes = sum(i is not None for i in [mag1_size, current_mag_size])
+        if num_bboxes == 0:
+            assert num_offsets != 0, "You must supply an offset or a bounding box."
+            assert num_sizes != 0, "You must supply a size."
```

Comment:
```suggestion
            assert num_sizes != 0, "You must supply a size or a bounding box."
```

Reply:
I'd leave it like this, since an offset is already given in this case, so adding a size seems more natural I guess? But no strong opinion.